solutions/Assignment-2.3/HttpServer.java

```
@@ -0,0 +1,200 @@
+/**
+ * JAVA code to implement a webserver that can handle basic GET requests.
+ * The HTTPServer waits for requests in the main thread and handles the requests
+ * in a separate ClientHandler thread.
+ */
+import java.io.*;
+import java.net.*;
+import java.util.*;
+import java.util.logging.*;
+
+// A Webserver waits for clients to connect, then starts a separate thread to handle the request.
+public class HttpServer {
+  private final static Logger logger = Logger.getLogger(HttpServer.class.getName());
+  private static FileHandler fileHandler = null;
+  private static ServerSocket serverSocket;
+  private static int port;
+  // Method to create log file
+  public static void initLog(){
+    try {
+      fileHandler = new FileHandler("Status.log", false);
+    }
+    catch (SecurityException | IOException e) {
+      e.printStackTrace();
+    }
+    Logger logger = Logger.getLogger("");
+    fileHandler.setFormatter(new SimpleFormatter());
+    logger.addHandler(fileHandler);
+    logger.setLevel(Level.FINE);
+  }
+
+  public static void main(String args[]) throws IOException {
+    initLog();
+    try {
+      port = Integer.parseInt(args[0]);
+      serverSocket = new ServerSocket(port); // Start,listen on given port
+      while (true) {
+        try {
+          logger.info("TCPServer Waiting for client on port " + port);
+          Socket socket = serverSocket.accept(); // Wait for a client to connect
+          new ClientHandler(socket); // A separate thread to handle each client
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+      }
+    } catch (NumberFormatException e) {
+        logger.warning("Port number should be a positive integer");
```

Once again, inconsistent indenting. But importantly, you're just ignoring this. I'd think it's better to raise the exception, since otherwise the calling method would have no idea that something was wrong.